src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnnecessaryParenthesesCheck.java

```diff
@@ -522,7 +534,11 @@ else if (TokenUtil.isOfType(type, ASSIGNMENTS)) {
                 assignDepth--;
             }
             else if (isSurrounded(ast) && unnecessaryParenAroundOperators(ast)) {
-                log(ast.getPreviousSibling(), MSG_EXPR);
+                DetailAST inParentheses = ast;
```

Comment:
does it make  sense to move new logic out of operators ?
https://docs.oracle.com/javase/tutorial/java/nutsandbolts/operators.html

you are targeting `.` 

so you code will be just another block `else if (TokenUtil.isOfType(type, ARRAY_AND_FIELD_ACCESS)) {`

Reply:
I thought your concern was about the special handling of `METHOD_CALL`, I refactored the code to make it more clear why that one token is handled differently.

There is not much of new logic here, the new tokens can be handled the same as the operators you linked, but maybe  `unnecessaryParenAroundOperators` should be renamed to `unnecessaryParenAroundExpressions` to be better aligned with Java terminology? Some languages consider `[` and `.` to be "operators" as well, but JLS talks about "array access expressions" and "field access expressions".